src/util/snippet.ts

```diff
@@ -0,0 +1,109 @@
+import { SnippetDefinition } from "../typings/snippet";
+import { Target } from "../typings/target.types";
+import {
+  Placeholder,
+  Text,
+  TextmateSnippet,
+  Variable,
+} from "../vendor/snippet/snippetParser";
+import { KnownSnippetVariableNames } from "../vendor/snippet/snippetVariables";
+
+/**
+ * Replaces the snippet variable with name `placeholderName` with
+ * TM_SELECTED_TEXT
+ *
+ * Also replaces any unknown variables with placeholders. We do this so it's
+ * easier to leave one of the placeholders blank. We may make it so that you can
+ * disable this with a setting in the future
+ * @param parsedSnippet The parsed textmate snippet to operate on
+ * @param placeholderName The variable name to replace with TM_SELECTED_TEXT
+ * @param substitutions A map from variable names to text values that will be
+ * substituted and the given variable will no longer be a placeholder in the
+ * final snippet
+ */
+export function transformSnippetVariables(
+  parsedSnippet: TextmateSnippet,
+  placeholderName?: string | null,
+  substitutions?: Record<string, string>
```

Comment:
This arg is new, for variable substitutions in insert, eg "snip funk hello world"